app/scraping.py

```diff
@@ -1,19 +1,48 @@
 from bs4 import BeautifulSoup
 import os
+import requests
+from PIL import Image
+from io import BytesIO
+import logging
 
+# Define global variables for minimum image dimensions
+MIN_IMAGE_WIDTH = int(os.getenv("MIN_IMAGE_WIDTH", 300))
+MIN_IMAGE_HEIGHT = int(os.getenv("MIN_IMAGE_HEIGHT", 250))
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
 def extract_images(soup: BeautifulSoup) -> list:
     images = []
     n_images = int(os.getenv("N_IMAGES", 4))  # Default to 4 if not set
+
     for img in soup.find_all(["img", "meta", "link", "source"]):
+        img_url = None
         if img.name == "img" and img.get("src"):
-            images.append(img["src"])
+            img_url = img["src"]
         elif img.name == "meta" and img.get("content") and "image" in img.get("property", ""):
-            images.append(img["content"])
+            img_url = img["content"]
         elif img.name == "link" and img.get("href") and "image" in img.get("type", ""):
-            images.append(img["href"])
+            img_url = img["href"]
         elif img.name == "source" and img.get("srcset"):
-            images.append(img["srcset"].split()[0])  # Take the first URL in srcset
+            img_url = img["srcset"].split()[0]  # Take the first URL in srcset
+
+        if img_url:
+            try:
+                response = requests.get(img_url)
+                response.raise_for_status()
+                image = Image.open(BytesIO(response.content))
+                if image.width >= MIN_IMAGE_WIDTH and image.height >= MIN_IMAGE_HEIGHT:
+                    images.append(img_url)
+            except requests.RequestException as e:
+                logger.error(f"RequestException for URL {img_url}: {e}")
```

Comment:
Seeing ERROR:app.scraping:RequestException for URL https://www.tob-dev.com/wp-content/uploads/2017/11/travel-to-bermuda.png: 406 Client Error: Not Acceptable for url: https://www.tob-dev.com/wp-content/uploads/2017/11/travel-to-bermuda.png

Reply:
Thank you for pointing that out. The 406 error indicates that the server is not accepting the request. This could be due to the server's configuration or the headers being sent with the request. I'll update the code to include appropriate headers to mimic a real browser request, which should help in resolving this issue. I'll make the necessary changes and push the updates shortly.